include/envelope.h

```diff
@@ -59,44 +59,49 @@
 #include <cstdint>
 #include <functional>
 
+
+typedef struct AudioFrame AudioFrame_;
```

Comment:
This is a new way (or style) we can or should start using to forward-declare types?

Yesterday @shermp investigated the build profile using https://github.com/nico/ninjatracing and https://ui.perfetto.dev/ to disocver that header parsing (at least on Windows) is chewing up the bulk of the time.

![unknown](https://user-images.githubusercontent.com/1557255/174448207-adfde4e9-6d98-4f6e-91e7-b05489e68730.png)

(but these are no doubt massive system headers laden with MMX and SSE inline intrinsics).

Reply:
> Is this is a new way (or style) we can (or should) start using to forward-declare types?

You put that in there 😄 and yes probably makes sense to forward declare (where possible). I found that it doesn't always work with `unique_ptr`s though.

I guess that's one of the big drawbacks of the lack of a proper module system in C++; you change a single line in one header file, then suddenly 80% of your project needs to be recompiled...